src/query/executor/transform/types.go

```diff
@@ -26,13 +26,61 @@ import (
 	"github.com/m3db/m3/src/query/block"
 	"github.com/m3db/m3/src/query/models"
 	"github.com/m3db/m3/src/query/parser"
+	"github.com/m3db/m3/src/x/instrument"
+	"github.com/pkg/errors"
 )
 
-// Options to create transform nodes
+var (
+	errNoInstrumentOptionsSet = errors.New("no instrument options set")
+)
+
+// Options to create transform nodes.
 type Options struct {
-	TimeSpec  TimeSpec
-	Debug     bool
-	BlockType models.FetchedBlockType
+	timeSpec          TimeSpec
+	debug             bool
+	blockType         models.FetchedBlockType
+	instrumentOptions instrument.Options
+}
+
+// OptionsParams are the params used to create Options.
+type OptionsParams struct {
+	TimeSpec          TimeSpec
+	Debug             bool
+	BlockType         models.FetchedBlockType
+	InstrumentOptions instrument.Options
+}
+
+// NewOptions enforces that fields are set when options is created.
+func NewOptions(p OptionsParams) (Options, error) {
```

Comment:
Yeah, it was possible to create it without `InstrumentOptions` and consequently it could easily lead to nil pointer panics.  So now it's protected against that by always enforcing it to be set when constructing `Options`.